### ISSUE TITLE
Dimacs and Delegates

### ISSUE THREAD

**Issue by ianamason** [SRI-CSL/yices2#257]
both deadlock in thread safe mode. Need to debug.

**Comment by ianamason** [comment]
OK so I fixed the deadlock, now I am checking for any races.  In the meantime I see complaints like this @BrunoDutertre :
```
Formulas:
  (= 0b00000011000000000001 (bv-mul x y))
  (= 0b00000100111000111111 (bv-mul y z))
  (= 0b00000010011111110011 (bv-mul x z))
==20707== Conditional jump or move depends on uninitialised value(s)
==20707==    at 0x4F43718: process_lit_equiv (new_sat_solver.c:6151)
==20707==    by 0x4F4383F: literal_equiv (new_sat_solver.c:6180)
==20707==    by 0x4F44E78: try_equivalent_vars (new_sat_solver.c:6897)
==20707==    by 0x4F4574A: pp_scc_simplification (new_sat_solver.c:7153)
==20707==    by 0x4F48751: nsat_cheap_preprocess (new_sat_solver.c:8370)
==20707==    by 0x4F487D2: nsat_preprocess (new_sat_solver.c:8395)
==20707==    by 0x4F4E293: nsat_do_preprocess (new_sat_solver.c:11046)
==20707==    by 0x4F4E587: nsat_solve (new_sat_solver.c:11159)
==20707==    by 0x4F31C8A: ysat_check (delegate.c:75)
==20707==    by 0x4F32990: solve_with_delegate (delegate.c:818)
==20707==    by 0x4E93266: check_with_delegate (context_solver.c:693)
==20707==    by 0x4E6B271: yices_do_check_formulas (yices_api.c:9589)
==20707==
==20707== Conditional jump or move depends on uninitialised value(s)
==20707==    at 0x4F43718: process_lit_equiv (new_sat_solver.c:6151)
==20707==    by 0x4F43829: literal_equiv (new_sat_solver.c:6178)
==20707==    by 0x4F43E38: process_lit_eq_ttbl (new_sat_solver.c:6525)
==20707==    by 0x4F44E9E: try_equivalent_vars (new_sat_solver.c:6903)
==20707==    by 0x4F4574A: pp_scc_simplification (new_sat_solver.c:7153)
==20707==    by 0x4F48751: nsat_cheap_preprocess (new_sat_solver.c:8370)
==20707==    by 0x4F487D2: nsat_preprocess (new_sat_solver.c:8395)
==20707==    by 0x4F4E293: nsat_do_preprocess (new_sat_solver.c:11046)
==20707==    by 0x4F4E587: nsat_solve (new_sat_solver.c:11159)
==20707==    by 0x4F31C8A: ysat_check (delegate.c:75)
==20707==    by 0x4F32990: solve_with_delegate (delegate.c:818)
==20707==    by 0x4E93266: check_with_delegate (context_solver.c:693)
==20707==
unsat

```

**Comment by BrunoDutertre** [comment]
Yes. valgrind is right. I'll push a fix shortly.

**Comment by BrunoDutertre** [comment]
Fixed read of uninitialized values in a2ae9b3aa7a0c6ba5eca9f6cb58438a0cf025d15.

**Comment by ianamason** [comment]
Verified. Also no races. closing by: 3c33394b4efb2b48696a7d441067f40b7a34d9cb
